app/javascript/util/listingUtil.ts

```diff
@@ -161,91 +169,205 @@ export const paperApplicationURLs = (isRental: boolean): PaperApplication[] => {
-//         mappedUnitsByOccupancy.push({
-//           occupancy: i,
-//           listingId: unit.listingID,
-//           absoluteMinIncome: unit.absoluteMinIncome,
-//           absoluteMaxIncome: unit.absoluteMaxIncome,
-//           summaryByAMI: [
-//             {
-//               unitMaxAMI: unit.unitMaxAMI,
-//               summaryByType: [{ ...unit }],
-//             },
-//           ],
-//         })
-//       } else {
-//         const ami = mappedOccupancy.summaryByAMI.find((s) => {
-//           return unit.unitMaxAMI === s.unitMaxAMI
-//         })
-
-//         if (ami) {
-//           ami.summaryByType.push({
-//             ...unit,
-//           })
-//         } else {
-//           mappedOccupancy.summaryByAMI.push({
-//             unitMaxAMI: unit.unitMaxAMI,
-//             summaryByType: [{ ...unit }],
-//           })
-//         }
-//       }
-//     }
-//   })
-//   return mappedUnitsByOccupancy
-// }
-
-export const classifyPricingDataByOccupancy = (
-  units: RailsListingPricingTableUnit[]
-): MappedUnitsByOccupancy[] => {
-  const mappedUnitsByOccupancy = []
-  units.forEach((unit: RailsListingPricingTableUnit) => {
-    const mappedOccupancy = mappedUnitsByOccupancy.find((s) => {
-      return s.occupancy === unit.maxOccupancy
-    })
+export const deriveIncomeFromAmiCharts = (
+  unit: RailsUnit,
+  occupancy: number,
+  amiCharts: RailsAmiChart[]
+): number => {
+  if (!unit || !occupancy || !amiCharts) {
+    return null
+  }
 
-    if (!mappedOccupancy) {
-      mappedUnitsByOccupancy.push({
-        occupancy: unit.maxOccupancy,
-        listingId: unit.listingID,
-        absoluteMinIncome: unit.absoluteMinIncome,
-        absoluteMaxIncome: unit.absoluteMaxIncome,
-        summaryByAMI: [
-          {
-            unitMaxAMI: unit.unitMaxAMI,
-            summaryByType: [{ ...unit }],
-          },
-        ],
-      })
-    } else {
-      const ami = mappedOccupancy.summaryByAMI.find((s) => {
-        return unit.unitMaxAMI === s.unitMaxAMI
-      })
+  const amiFromAmiChart = amiCharts.find((amiData: RailsAmiChart) => {
+    return (
+      Number(amiData.percent) === unit.Max_AMI_for_Qualifying_Unit &&
+      amiData.chartType === unit.AMI_chart_type
+    )
+  })?.values
+
+  const occupancyAmi = amiFromAmiChart?.find((ami: RailsAmiChartValue) => {
+    return ami.numOfHousehold === occupancy
+  })
+
+  if (occupancyAmi) {
+    return Math.floor(occupancyAmi?.amount / 12)
+  }
+
+  return null
+}
+
+export const applyMaxIncomeToUnit = (amiCharts: RailsAmiChart[]) => (
+  unit: RailsUnitWithOccupancy
+): RailsUnitWithOccupancyAndMaxIncome => {
+  const maxMonthlyIncomeNeeded = deriveIncomeFromAmiCharts(unit, unit.occupancy, amiCharts)
+  return { ...unit, maxMonthlyIncomeNeeded }
+}
+
+export const addUnitsWithEachOccupancy = (units: RailsUnit[]): RailsUnitWithOccupancy[] => {
+  const totalUnits = []
+  units.forEach((unit: RailsUnit) => {
+    if (!unit.Max_Occupancy) {
+      unit.Max_Occupancy = 3
```

Comment:
good question! ha no idea tbh, it's what most of the sales listings seem to be set to in angular. I couldn't find any spec around it : (. Something I decided to do since it mostly makes it correct and then see what the product team has to say about it. That's a terrible answer I'm so sorry. How do you feel about a code comment there?